IOMC/Input/src/MCFileSource.cc

```diff
@@ -32,7 +33,8 @@ MCFileSource::MCFileSource(const ParameterSet & pset, InputSourceDescription con
   }  
   
   reader_->initialize(fileName);  
-  produces<HepMCProduct>();
+  produces<HepMCProduct>("generator");
```

Comment:
if I understand correctly your question, the module concerned by this PR needs to run before the vertex smearing as is needed to import into EDM format a showered set of events in HepMC format